File: csharp/grains/Grains.cs
using System;
using System.Linq;

public static class Grains
{
    public static double Square(int i)
    {
        if (i <= 0 || i > 64)
        {
            throw new ArgumentOutOfRangeException(nameof(i));
        }
        
        return Math.Pow(2, i - 1);
    }

    public static double Total() => Enumerable.Range(1, 64).Select(Square).Sum();
}

File: typescript/grains/grains.ts
function* range(begin: number, end: number) {
    for (let i = begin; i <= end; i++) {
        yield i
    }
}

export default class Grains {
    public static square(n: number) {
        if (n < 1 || n > 64) {
            return -1
        }

        return 2 ** (n - 1)
    }

    public static total() {
        return Array.from(range(1, 64), this.square).reduce((x, y) => x + y)
    }
}